permissions_validators/src/private_blockchain/mod.rs

```diff
@@ -38,3 +38,541 @@ impl<W: WorldTrait> IsGrantAllowed<W> for ProhibitGrant {
+            let mut bob = Account::new(bob_id.clone(), []).build();
+
+            // Silver asset
+            let silver_asset_definition_id =
+                AssetDefinitionId::from_str("silver#test0").expect("Valid");
+            let silver_asset_id = AssetId::new(silver_asset_definition_id.clone(), bob_id.clone());
+            let silver_asset = Asset::new(silver_asset_id.clone(), AssetValue::Quantity(200));
+
+            // Bob owns Silver
+            bob.add_asset(silver_asset);
+
+            // Carol
+            let carol_id = AccountId::from_str("carol@test1").expect("Valid");
+            let mut carol = Account::new(carol_id.clone(), []).build();
+
+            // Bronze asset
+            let bronze_asset_definition_id =
+                AssetDefinitionId::from_str("bronze#test1").expect("Valid");
+            let bronze_asset_id =
+                AssetId::new(bronze_asset_definition_id.clone(), carol_id.clone());
+            let bronze_asset = Asset::new(bronze_asset_id.clone(), AssetValue::Quantity(300));
+
+            // Carol owns Bronze
+            carol.add_asset(bronze_asset.clone());
+
+            // Alice also owns Bronze
+            alice.add_asset(bronze_asset);
+
+            // First domain
+            let first_domain_id = DomainId::from_str("test0").expect("Valid");
+            let mut first_domain = Domain::new(first_domain_id.clone()).build();
+
+            // First domain includes Alice
+            first_domain.add_account(alice);
+
+            // First domain includes Bob
+            first_domain.add_account(bob);
+
+            // Second domain
+            let second_domain_id = DomainId::from_str("test1").expect("Valid");
+            let mut second_domain = Domain::new(second_domain_id.clone()).build();
+
+            // Second domain includes Carol
+            second_domain.add_account(carol);
+
+            let wsv = WorldStateView::<World>::new(World::with(
+                [first_domain.clone(), second_domain.clone()],
+                vec![],
+            ));
+
+            Self {
+                alice_id,
+                bob_id,
+                carol_id,
+                gold_asset_id,
+                gold_asset_definition_id,
+                silver_asset_id,
+                silver_asset_definition_id,
+                bronze_asset_id,
+                bronze_asset_definition_id,
+                first_domain: (first_domain_id, first_domain),
+                second_domain: (second_domain_id, second_domain),
+                wsv,
+            }
+        }
+    }
+
+    #[test]
+    fn find_all_accounts() {
+        let TestEnv {
+            alice_id,
+            bob_id,
+            carol_id,
+            wsv,
+            ..
+        } = TestEnv::new();
+
+        let op = QueryBox::FindAllAccounts(FindAllAccounts::new());
+
+        {
+            let only_accounts_domain: IsQueryAllowedBoxed<World> = query::OnlyAccountsDomain.into();
+
```

Comment:
There should be a negative test-case (with `is_ok()`).

Reply:
Why not to use another *query* to check `is_ok()` variant?

Reply:
Oh I see, it tests queries, not validators